src/main/java/htsjdk/samtools/SAMFileWriterFactory.java

```diff
@@ -363,42 +363,61 @@ private SAMFileWriter initWriter(final SAMFileHeader header, final boolean preso
     }
 
     /**
-     * Create either a SAM or a BAM writer based on examination of the outputFile extension.
+     * Create either a SAM or a BAM writer based on examination of the outputFile extension. The method fails
+     * with {@link IllegalArgumentException} if the outputFile extension is neither .sam not .bam.
      *
      * @param header     entire header. Sort order is determined by the sortOrder property of this arg.
      * @param presorted  presorted if true, SAMRecords must be added to the SAMFileWriter in order that agrees with header.sortOrder.
      * @param outputFile where to write the output.  Must end with .sam or .bam.
      * @return SAM or BAM writer based on file extension of outputFile.
+     * @throws IllegalArgumentException if the file extension is neither .sam nor .bam.
      */
     public SAMFileWriter makeSAMOrBAMWriter(final SAMFileHeader header, final boolean presorted, final File outputFile) {
```

Comment:
What happens if the file is `/dev/stdout` or `/dev/stderr`?  Previously it worked, now it likely doesn't.  What am I missing?